### PR TITLE
Fixed ec2_group_info module to work with Python 3.8

### PR DESCRIPTION
**Pull request by stefanhorning**
…without modifying the dict in loop (immutable).

##### SUMMARY
Fixes Python runtime error `dictionary keys changed during iteration`

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ec2_group_info module

##### ADDITIONAL INFORMATION
Output of ansible --version (on Ubuntu 20.04):
```
ansible 2.9.1
  config file = /xx/ansible.cfg
  configured module search path = ['/xx/library']
  ansible python module location = /xx/.local/lib/python3.8/site-packages/ansible
  executable location = /xx/.local/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

Error output of module without the fix:
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: RuntimeError: dictionary keys changed during iteration
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 102, in <module>\n  File \"<stdin>\", line 94, in _ansiballz_main\n  File \"<stdin>\", line 40, in invoke_module\n  File \"/usr/lib/python3.8/runpy.py\", line 207, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.8/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_ec2_group_info_payload_80im7jvm/ansible_ec2_group_info_payload.zip/ansible/modules/cloud/amazon/ec2_group_info.py\", line 165, in <module>\n  File \"/tmp/ansible_ec2_group_info_payload_80im7jvm/ansible_ec2_group_info_payload.zip/ansible/modules/cloud/amazon/ec2_group_info.py\", line 142, in main\nRuntimeError: dictionary keys changed during iteration\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

Which goes away with this patch.

**NOTE:** Perhaps a test with Python 3.8 should be added to testsuite.

### CODE REVIEW

File: plugins/modules/ec2_group_info.py [hunk 1]
```diff
@@ -115,10 +115,12 @@ def main():
...
     # Replace filter key underscores with dashes, for compatibility, except if we're dealing with tags
-    sanitized_filters = module.params.get("filters")
-    for key in list(sanitized_filters):
+    filters = module.params.get("filters")
+    sanitized_filters = dict()
+
+    for key in filters:
         if not key.startswith("tag:"):
-            sanitized_filters[key.replace("_", "-")] = sanitized_filters.pop(key)
+            sanitized_filters[key.replace("_", "-")] = filters[key]
```
Comment (tremble) [inline plugins/modules/ec2_group_info.py:123]: This looks like it's going to drop all 'tag:' filters on the floor...

**Comment by stefanhorning** [comment]
Build failed due to timeout issues when downloading stuff (unrelated to my change).

**Review by tremble — CHANGES_REQUESTED** [review]
Hi,

thanks for taking the time to open this pull request.

A couple of things:
- We currently don't have any integration tests for ec2_group_info, would you be willing to update the ec2_group tests to include some tests for ec2_group_info (update the playbook and add 'ec2_group_info' to the aliases file)?
- Please add a changelog entry: https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to

While we don't currently run integration tests against 3.8 we do run them against 3.7 which *should* have caught this, unfortunately ec2_group_info is currently missing from our integration test suite.

**Comment by stefanhorning** [comment]
>     * We currently don't have any integration tests for ec2_group_info, would you be willing to update the ec2_group tests to include some tests for ec2_group_info (update the playbook and add 'ec2_group_info' to the aliases file)?

I noticed the same thing and was thinking the same. Can tackle this tomorrow I think.

File: plugins/modules/ec2_group_info.py (hunk 1 above)
Comment (stefanhorning) [reply plugins/modules/ec2_group_info.py:123]: Ah true :facepalm:

File: tests/integration/targets/ec2_group/tasks/group_info.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        name: '{{ ec2_group_name }}-info-1'
+        vpc_id: '{{ vpc_result.vpc.id }}'
+        description: '{{ ec2_group_description }}'
+        rules:
+          - proto: tcp
+            ports:
+              - 90
+            cidr_ip: 10.2.2.2/32
+        tags:
+          - test: ec2_group_info_module
```
Comment (tremble) [inline tests/integration/targets/ec2_group/tasks/group_info.yml:18]:
```suggestion
          - test: '{{ test_prefix }}ec2_group_info_module'
```

About 6 copies of the test will run in parallel, searching based on the tag will generate unpredictable results.

Comment (stefanhorning) [reply tests/integration/targets/ec2_group/tasks/group_info.yml:18]: Ok. Fastest response ever ;)

File: changelogs/fragments/181-ec2-group-info-python-fix.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - Code fix in ec2_group_info so module works with Python 3.8 (make dict immutable in loop)
```
Comment (tremble) [inline changelogs/fragments/181-ec2-group-info-python-fix.yaml:3]:
```suggestion
  - ec2_group_info - Code fix so module works with Python 3.8 (make dict immutable in loop) (https://github.com/ansible-collections/amazon.aws/pull/181)
```

Formatting / style niggle

File: tests/integration/targets/ec2_group/tasks/group_info.yml (hunk 2 above)
Comment (tremble) [reply tests/integration/targets/ec2_group/tasks/group_info.yml]: Luck of the draw.

You marked your initial commit as a draft.  I'll wait for a comment from you that you're ready before I start looking at how thorough the tests are, but searching for static strings has caused me enough pain that I didn't want you to stare at failures wondering WTF is going on.

Comment (stefanhorning) [reply tests/integration/targets/ec2_group/tasks/group_info.yml]: Yep, thanks for that! However `test_prefix` doesn't seem to be defined. Do I have to set this variable up myself somehow?

Comment (stefanhorning) [reply tests/integration/targets/ec2_group/tasks/group_info.yml]: I guess it should be `resource_prefix` instead, will try that.

**Comment by stefanhorning** [comment]
Ready for final review now.

**Review by tremble — APPROVED** [review]
Looks good, thanks for taking the time to write those tests.

**Pull request merged**